keras_cv/models/object_detection/retina_net/retina_net.py

```diff
@@ -25,28 +25,29 @@
 from keras_cv.models.object_detection.retina_net.__internal__ import (
     layers as layers_lib,
 )
+from keras_cv.utils.train import get_feature_extractor
 
 BOX_VARIANCE = [0.1, 0.1, 0.2, 0.2]
 
 
 # TODO(lukewood): update docstring to include documentation on creating a custom label
 # decoder/etc.
-# TODO(lukewood): link to keras.io guide on creating custom backbone and FPN.
+# TODO(jbischof): Generalize `FeaturePyramid` class to allow for any P-levels and
+# add `feature_pyramid_levels` param.
 @keras.utils.register_keras_serializable(package="keras_cv")
 class RetinaNet(tf.keras.Model):
     """A Keras model implementing the RetinaNet architecture.
 
     Implements the RetinaNet architecture for object detection.  The constructor
-    requires `classes`, `bounding_box_format` and a `backbone`.  Optionally, a
-    custom label encoder, feature pyramid network, and prediction decoder may all be
-    provided.
+    requires `classes`, and `bounding_box_format`.  Optionally, a backbone,
```

Comment:
One thing we need to decide once across all of our packages is whether to use `classes` or `num_classes`. Right now I believe we have a bit of both across different APIs, which is inconsistent. We should use consistent naming everywhere.

Reply:
Yes, I already got buy-in and will have a follow PR to migrate everything to `num_classes`

Reply:
Opened #1473